api/envoy/http/common/base.proto

```diff
@@ -65,7 +65,8 @@ message HttpUri {
       [(validate.rules).duration.required = true];
 }
 
-/* TODO(taoxuy): replace with envoy internal DataSource proto */
+// Duplicate DataSource from envoy for self containment.
+// https://aip.dev/213
```

Comment:
Maybe you can try to see if we can use Envoy api DataSource directly instead of defining our own.
I know for now, CloudESF api doens't depends on any Envoy api.  Maybe we can change it.

Can be a separate as clean up tech debt

Reply:
Same for HttpUri,

Reply:
@qiwzhang I disagree, it's harder to version our APIs if they're tied to envoy.

From https://aip.dev/213

> APIs **must not** define a set of API-specific common protos which live outside of its versioning structure. This prevents independent movement of particular versions ...

Given that Datasource is a versioned envoy proto, this would be a breaking change if we update from v3->v4, etc. While we can handle breaking changes, it just seems like more maintenance work for us.

Reply:
OK, good point. I am convinced. Thanks.